lang/ruby/lib/avro/schema_compatibility.rb

```diff
@@ -141,6 +141,8 @@ def match_union_schemas(writers_schema, readers_schema)
       end
 
       def match_record_schemas(writers_schema, readers_schema)
+        return false if writers_schema.type_sym == :union
```

Comment:
Haven't done Ruby for a while. This will return always `false` when it is a union?

Reply:
That's correct it will always return `false` for a union.

From a schema compatibility perspective, that is correct. If there writer's schema was union, then in general it cannot be guaranteed that a record schema in the reader is compatible with it.